commands/utils/docker-endpoint.ts

```diff
@@ -9,14 +9,8 @@ class DockerClient {
     private endPoint:Docker;
 
     constructor() {
-        if (process.env.DOCKER_HOST){
-            this.endPoint = new Docker();
-        }
-        else if (process.platform === 'win32') {
-            this.endPoint = new Docker({ socketPath: "//./pipe/docker_engine" });
-        } else {
-            this.endPoint = new Docker({ socketPath: '/var/run/docker.sock' });
-        }
+        // Pass no options so that the defaultOpts of docker-modem will be used
+        this.endPoint = new Docker(null);
```

Comment:
Yes I did not like it either but I did it to keep the ts compiler from complaining.

The problem seems to be with the embedded typings file that contain the following offending line.
`constructor(options: Docker.DockerOptions);`

The options argument should be optional. Unfortunately I do not know the place where this file was sourced as I have seen other implementations fixing that problem. If we change the typings file we can delete the null argument.